File: sonarqube-companion-rest/src/main/java/pl/consdata/ico/sqcompanion/history/ViolationsHistory.java
package pl.consdata.ico.sqcompanion.history;

import lombok.Builder;
import lombok.Data;
import lombok.Singular;

import java.util.List;

@Data
@Builder
public class ViolationsHistory {

    @Singular
    private List<ViolationHistoryEntry> violationHistoryEntries;

}

File: sonarqube-companion-frontend/src/app/violations/violations-history.ts
import {ViolationsHistoryEntry} from './violations-history-entry';

export class ViolationsHistory {

  history: ViolationsHistoryEntry[];

  constructor(data) {
    this.history = data.violationHistoryEntries.map(entry => new ViolationsHistoryEntry(entry));
  }

  isEmpty(): boolean {
    return !this.history || this.history.length === 0;
  }

}
